components/post_markdown/index.ts

```diff
@@ -31,17 +37,34 @@ export function makeGetMentionKeysForPost() {
     );
 }
 
+type OwnProps = {
+    channelId?: string;
```

Comment:
This is a little bit complicated. 

in the `Textbox` component, we called the `PostMarkdown` and we don't pass the `post`. so the `getMentionKeyForPost` has a type error since the post is required. so if I make the `post` optional I should change the `makeGetMentionKeysForPost`. instead i change the logic in `mapStateToProps`.


also in the Textbox component, the `channelId` prop is optional so the `channelId` here may be `undefined`. 

about the mentionKeys, some components are providing mentionKeys as props and we also make mention key in mapStateToProps so which one should be passed to the component.

Reply:
In `getMentionKeysForPost`, we are already coalescing on the `post` object in case it is null. If it is, we return just the value of `getCurrentUserMentionKeys`. So I think the `post` parameter in that selector should be optional and we should restore the logic in `mapStateToProps` to what it originally was.

As for `channelId`, every use of `Textbox` currently passes in `channelId`, so it should always be defined. It's possible when that component was converted that we should have made `channelId` required, but in this case it might make sense to simply assert the channelId parameter as passed into `PostMarkdown` from `Textbox` so that it's always defined.